lib/Dinky.d.ts

```diff
@@ -0,0 +1,821 @@
+   *
+   * @param list – a tag or a list of tags you want to append
+   */
+  query(...list: (string | string[])[]): this
+
+  /**
+   * Adds a param to sort result by given field.
+   *
+   * @param field Filed name to sort by. Note that sorting doesn't work for all fields.
+   *  Consult the [api docs](https://derpibooru.org/pages/api) to learn more
+   */
+  sortBy(field: string): this
+}
+
+interface ImagesSearch extends Search<responses.ImagesResponse> {
+  /**
+   * Sets `my:faves` param to the search request.
+   *
+   * Note that this method requires user's key.
+   */
+  faves(): this
+
+  /**
+   * Sets `my:watched` param to the search request.
+   *
+   * Note that this method requires user's key.
+   */
+  watched(): this
+
+  /**
+   * Sets `my:upvotes` param to the search request.
+   *
+   * Note that this method requires user's key.
+   */
+  upvotes(): this
+
+  /**
+   * Sets `my:downvotes` param to the search request.
+   *
+   * Note that this method requires user's key.
+   */
+  downvotes(): this
+
+  /**
+   * Sets `my:uploads` param to the search request.
+   *
+   * Note that this method requires user's key.
+   */
+  uploads(): this
+
+  /**
+   * Search for the images faved by given user.
+   *
+   * @param user – name of the user on Derpibooru
+   */
+  favedBy(user: string): this
+
+  /**
+   * Search for images uploaded by specified user
+   *
+   * @param user – name of the user on Derpibooru
+   */
+  uploadedBy(user: string): this
+
+  /**
+   * Specifies how many images per page should API return.
+   *
+   * @param value – an amount of images you want to take.
+   *   The value must be in range of `1` and `50`.
+   */
+  limit(value: number): this
+
+  /**
+   * Sets the MINIMAL score of requested images.
+   *
+   * @param value – a value of minimal score
+   */
+  minScore(value: number): this
+
+  /**
+   * Sets the MAXIMAL score of requested images.
+   *
+   * @param value – a value of maximal score
+   */
+  maxScore(value: number): this
+
+  /**
+   * Sets images ordering to ascending.
+   */
+  ascending(): this
+
+  /**
+   * Sets images ordering to descending.
+   */
+  descending(): this
+
+  /**
+   * If been called, the API will return random image.
+   */
+  random(): this
+}
+
+/**
+ * Contains interface declarations of all the API responses that are sent back
+ * from derpibooru. Note that though the original reponses use snake_case
+ * properties naming, in TypeScript `camelCase` is more conventional, so these are
+ * transformed to `camelCase` in `dinky.js`
+ */
+namespace responses {
+  // There are no strict schema declarations, the types
+  // are manually written using https://derpibooru.org/pages/api
+
+  interface WithTotal {
+    /**
+     * Total number of entities that can be retreived by paginating through them.
+     */
+    total: number
+  }
+
+  /**
+   * String in RFC3339 datetime format
+   */
+  export type DateTimeString = string;
```

Comment:
Btw, dates will be returned as Date, not as a string
See: https://github.com/octet-stream/dinky/blob/bf560fd3f99afd7285b100d721b71f6c50e3ab55/lib/util/castDates.js#L20

Reply:
Hmm, yeah `Date` type should go there instead